[Dev] Update documentation for the localization process (#705)
Despair-Games/poketernity: README.md

- node: 20.13.1
- npm: [how to install](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)
- git: [how to install](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) (needed to retrieve the translations)

#### Running Locally

> 1. Clone the repo through git and in the root directory run `npm install`

Reviewer: ```suggestion
1. Clone the repo through git and in the root directory run `npm install`.
```

If the install process seems to hang at the post-install step, you likely just need to press `Enter` for it continue.

> <!--   - _if you run into any errors, reach out in the **#dev-corner** channel in discord_-->

Reviewer 2: ```suggestion
<!--   - _if you run into any errors, reach out in [TBD]_-->
```

> 2. Run `npm run start:dev` to locally run the project in `localhost:8000`

Reviewer: ```suggestion
2. Run `npm run start:dev` to locally run the project in `localhost:8000`.
```

> 2. Run `npm run start:dev` to locally run the project in `localhost:8000`.

Reviewer 2: ```suggestion
2. Run `npm run start:dev` to locally run the project. You can then access it from the same machine by putting `http://localhost:8000/` in your browser's address bar.
```

> If you want to access the game from any device on your local network, you can run `npm run start:dev -- --host`

Author: ```suggestion
If you want to access the game from any device on your local network, you can run `npm run start:dev -- --host`.
```

> If you want to access the game from any device on your local network, you can run `npm run start:dev -- --host`.

Reviewer 2: ```suggestion
If you want to access the game from other devices on your local network, you can run `npm run start:dev -- --host` instead. You can then connect via `http://[IP of your device]:8000/` (ie: `http://192.168.1.101:8000/`).
```

#### Tests

We are using [vitest](https://vitest.dev/) as a testing framework for the game. The full test suite can be run with the command `npm run test:silent`. Most PRs are expected to add tests for their new features or bug fixes to avoid future regression. A basic test file for a variety of cases can be created by running the `npm run test:create` command.

#### Code-Style

> 

Reviewer 2: I can't leave a suggestion on the line I want so I'll put it one above:
`prettier` -> `Prettier`

- node: 20.13.1
- npm: [how to install](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)
- git: [how to install](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) (needed to retrieve the translations)

#### Running Locally

1. Clone the repo through git and in the root directory run `npm install`.
If the install process seems to hang at the post-install step, you likely just need to press `Enter` for it continue.
<!--   - _if you run into any errors, reach out in [TBD]_-->
2. Run `npm run start:dev` to locally run the project. You can then access it from the same machine by putting `http://localhost:8000/` in your browser's address bar.

> If you want to access the game from other devices on your local network, you can run `npm run start:dev -- --host` instead. You can then connect via `http://[IP of your device]:8000/` (ie: `http://192.168.1.101:8000/`).

Reviewer: ```suggestion
If you want to access the game from other devices on your local network, you can run `npm run start:dev -- --host` instead. You can then connect via `http://[IP of your device]:8000/` (e.g.: `http://192.168.1.101:8000/`).
```

Reviewer 2: You added an extra comma in the suggestion.

#### Tests

We are using [Vitest](https://vitest.dev/) as a testing framework for the game. The full test suite can be run with the command `npm run test:silent`. Most PRs are expected to add tests for their new features or bug fixes to avoid future regression. A basic test file for a variety of cases can be created by running the `npm run test:create` command.

> 

Reviewer 2: ```suggestion
You can also use `npm run typecheck` to invoke the Typescript compiler to check for basic code errors.

```

Author: thank you. Added a whole section regrouping this and depcruise + typedoc commands. let me know if it seems fine

Reviewer 2: Looks good. 👍

- node: 20.13.1
- npm: [how to install](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)
- git: [how to install](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) (needed to retrieve the translations)

#### Running Locally

> 1. Clone the repo through git and in the root directory run `npm install`.

Reviewer 3: Should this also say to clone the locales repo as well?

Author: no, that's all handled automatically by npm install in the post install part (just need to press enter)

#### Tests

We are using [Vitest](https://vitest.dev/) as a testing framework for the game. The full test suite can be run with the command `npm run test:silent`. Most PRs are expected to add tests for their new features or bug fixes to avoid future regression. A basic test file for a variety of cases can be created by running the `npm run test:create` command.

#### Code-Style

We are using [Prettier](https://prettier.io/) to format our code. It will run automatically during the pre-commit hook so don't worry about having to format the code manually properly.

#### Linting

We're using [ESLint](https://eslint.org/docs/latest/rules/) plus the [ESLint Stylistic](https://eslint.style/rules) and [Typescript ESLint](https://typescript-eslint.io/rules/) plugins for linting. It will run automatically via the pre-commit hook, but if you would like to run it manually you can use the `npm run eslint` script. To view the currently applied ESLint rules, check out the [eslint.config.js](./eslint.config.js) file.

#### Continuous Integration

> Github Workflows are used on every PR to enforce the test suite being successful, proper linting, no compilation errors and no circular dependencies in the codebase.

Reviewer 4: should `npm run test` be included here as well?

Reviewer 2: I think it makes sense to combine the "Tests" and "CI" sections into one.

Author: Does this [new version](https://github.com/Despair-Games/poketernity/blob/7daa5e726bfffa3a8b63fcf794b258e6808ab062/README.md) make sense?

Author: <sub> also i realized there is no mention of Phaser anywhere but don't know if I have the motivation to add that in this PR

> We're using ESLint as our common linter and formatter. It will run automatically during the pre-commit hook but if you would like to manually run it, use the `npm run eslint` script. To view the complete rules, check out the [eslint.config.js](./eslint.config.js) file.

Reviewer 2: ```suggestion
We're using [ESLint](https://eslint.org/docs/latest/rules/) plus the [ESLint Stylistic](https://eslint.style/rules) and [Typescript ESLint](https://typescript-eslint.io/rules/) plugins for linting. It will run automatically via the pre-commit hook, but if you would like to run it manually you can use the `npm run eslint` script. To view the currently applied ESLint rules, check out the [eslint.config.js](./eslint.config.js) file.
```

**How do I retrieve the translations?**

- The translations are found in a [dedicated repository](https://github.com/despair-games/poketernity-locales) and are applied as a submodule in this project.
- The command to retrieve the translations is `git submodule update --init --recursive`. <!--If you still struggle to get it working, please reach out in [TBD].-->

> [!NOTE]

> > We have recently made major changes to our testing framework (see #158 and #247). If you experience any weird issues with unit tests, especially issues that only occur when running multiple test files in a single run, please feel free to reach out for help.

Reviewer 2: This should be in the "Tests" section (and maybe isn't necessary any more anyway? The test suite seems pretty stable).
